hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/SQLiteDialect.java

```diff
@@ -457,9 +457,9 @@ public ViolatedConstraintNameExtractor getViolatedConstraintNameExtractor() {
 
 	private static final ViolatedConstraintNameExtractor EXTRACTOR =
 			new TemplatedViolatedConstraintNameExtractor( sqle -> {
-				final int errorCode = JdbcExceptionHelper.extractErrorCode( sqle );
+				final int errorCode = JdbcExceptionHelper.extractErrorCode( sqle ) & 0xFF;
 				if (errorCode == SQLITE_CONSTRAINT) {
-					return extractUsingTemplate( "constraint ", " failed", sqle.getMessage() );
+					return extractUsingTemplate( "constraint failed: ", "$", sqle.getMessage() );
```

Comment:
If there's no end to the constraint name within the given message might as well use `\n` to avoid calling `message.indexOf()`.

```suggestion
					return extractUsingTemplate( "constraint failed: ", "\n", sqle.getMessage() );
```

Reply:
Indeed !

Reply:
Should I commit your suggestion ?

Reply:
Sure! Please squash everything to a single commit when you're done.